assembly/index.ts

```diff
@@ -60,21 +60,14 @@ export function write_signed_leb128_u32(ptr: u32, value: u32): u32 {
 
 @inline
 export function leb128_u32_byte_size(v: u32): u32 {
-  let wrote: u32 = 0;
-  while (v >= 0x80) {
-    wrote += 1
-    v >>= 7;
-  }
-  wrote += 1
-  return wrote;
+  return 8 - (clz(v) + 31) / 8;
 }
 
 // Write a Vec of a given size to `ptr`. Arguments are used instead of an array
 // because the array requires runtime support for allocations.
 @inline
 export function write_vec5<T>(ptr: u32, arg1: T, arg2: T, arg3: T, arg4: T, arg5: T): u32 {
-  let wrote: u32 = 0;
-  wrote += write_leb128_u32(ptr, 5);
+  let wrote: u32 = write_leb128_u32(ptr, 5);
```

Comment:
Also, it could be simply replaced to `let wrote = write_u8(ptr, 5);` due to LEB128 with `v <= 127` equivalent to single byte